cluster-autoscaler/utils.go

```diff
@@ -271,3 +271,15 @@ func fixNodeGroupSize(context *AutoscalingContext, currentTime time.Time) (bool,
 	}
 	return fixed, nil
 }
+
+// Enables cloud provider specific node protection for each node group.
+func protectNodes(context *AutoscalingContext) error {
+	for _, group := range context.CloudProvider.NodeGroups() {
```

Comment:
Can we just check node group type, cast and don't add this AWS-specific feature to every cloud provider?